src/net_processing.cpp

```diff
@@ -2021,8 +2021,8 @@ static void ProcessHeadersMessage(CNode& pfrom, CConnman& connman, ChainstateMan
         if (!pfrom.fDisconnect && pfrom.IsFullOutboundConn() && nodestate->pindexBestKnownBlock != nullptr) {
             // If this is an outbound full-relay peer, check to see if we should protect
             // it from the bad/lagging chain logic.
-            // Note that block-relay-only peers are already implicitly protected, so we
-            // only consider setting m_protect for the full-relay peers.
+            // Note that outbound block-relay peers are excluded from this protection, and
```

Comment:
This code is in the conditional `&& pfrom.IsFullOutboundConn()`.

Why comments about `block-relay-only` connections belong in a block about `full conns` at all?

Reply:
I would probably prefer to have a good big comment *before* the following line:
```
         if (!pfrom.fDisconnect && pfrom.IsFullOutboundConn() && nodestate->pindexBestKnownBlock != nullptr) {
```

Reply:
Moved comment above with new referral to ChainSyncTimeout.